### PR TITLE
Return an exit code from sky_shell representing what kind of error occurred

### PR DESCRIPTION
**Pull request by jason-simmons**
This is intended to match the exit codes returned by the Dart command line tool

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @a-siva

File: runtime/dart_controller.cc [hunk 1]
```diff
@@ -120,15 +120,18 @@ void DartController::RunFromSnapshot(const uint8_t* buffer, size_t size) {
...
   tonic::DartState::Scope scope(dart_state());
   tonic::FileLoader& loader = dart_state()->file_loader();
   if (!packages.empty() && !loader.LoadPackagesMap(ResolvePath(packages)))
     FTL_LOG(WARNING) << "Failed to load package map: " << packages;
-  LogIfError(loader.LoadScript(main));
+  Dart_Handle result = loader.LoadScript(main);
+  LogIfError(result);
+  tonic::DartErrorHandleType error = tonic::GetErrorHandleType(result);
   if (SendStartMessage(Dart_RootLibrary()))
     exit(1);
```
Comment (chinmaygarde) [inline runtime/dart_controller.cc]: Shouldn't we return kUnknownErrorType here?

File: shell/platform/linux/main_linux.cc [hunk 2]
```diff
@@ -17,32 +17,60 @@
...
 #include "flutter/shell/testing/testing.h"
 #include "flutter/sky/engine/public/web/Sky.h"
+#include "lib/tonic/dart_microtask_queue.h"
 
 namespace {
 
+// Exit codes used by the Dart command line tool.
+const int kApiErrorExitCode = 253;
+const int kCompilationErrorExitCode = 254;
+const int kErrorExitCode = 255;
```
Comment (Hixie) [inline shell/platform/linux/main_linux.cc]: Aren't POSIX exit codes 7 bits?

Comment (Hixie) [inline shell/platform/linux/main_linux.cc]: ...in particular, how will this interact with the code in flutter's test harness that checks for segfaults using the exit code

File: runtime/dart_controller.cc (hunk 1 above)
Comment (jason-simmons) [inline runtime/dart_controller.cc]: For now I kept the existing behavior.

All of the DartController::Run methods are exiting the process if SendStartMessage fails.  I'm not sure why this was done - are we concerned that we can't cleanly shut down the engine and Dart VM if there's an error at startup?

File: shell/platform/linux/main_linux.cc (hunk 2 above)
Comment (jason-simmons) [inline shell/platform/linux/main_linux.cc]: The manpage says that the lower 8 bits of the exit code are used

**Pull request merged**